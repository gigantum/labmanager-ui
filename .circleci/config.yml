version: 2
jobs:
  build:
    docker:
      - image: gigantum/circleci-service-client:992bcbab-2018-07-09
    steps:
      - checkout
      - setup_remote_docker
      - run:
          name: Initialize submodules
          command: |
            git submodule update --init --recursive
      - run:
          name: Build frontend production code
          command: |
            npm install
            npm install relay
            npm run relay
            unset CI && npm run build
      - run:
          name: Initialize submodules
          command: |
            jest dashboard -u
