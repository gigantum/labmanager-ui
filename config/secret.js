const secrets = {
  accessToken: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik56a3dRekV5TVRORFJVTXpRakkzUmtRMlFrVkVRME0xT0RCRE5qRXhNalJHT1VOQ1JEQkJRZyJ9.eyJpc3MiOiJodHRwczovL2dpZ2FudHVtLmF1dGgwLmNvbS8iLCJzdWIiOiJhdXRoMHw1OWU4YzgyYmMzMGEzODA1M2FiNTM2YjMiLCJhdWQiOlsiYXBpLmdpZ2FudHVtLmlvIiwiaHR0cHM6Ly9naWdhbnR1bS5hdXRoMC5jb20vdXNlcmluZm8iXSwiaWF0IjoxNTI3MDg4Mzg1LCJleHAiOjE1MjcxNzQzODUsImF6cCI6IjBBam84T3Y2UXJpYWZvMHhWcFN5ZWpXeTFwd05XZGhrIiwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCJ9.QHxIdCtoHr1DgAmyYhxQ70J2p0j0HFBDAuCPPT_Vr6yVFsyJNMvYLivedvgM7HJ2WcSgc84ZEeaG95SNkTl21wA4GnpP2vLJc82DecFm-iDM5ip-6fvW9-Gr0HTRIldfAe-qtiLy70DMbBuQW7zqEMAs8FHIn8zqPpG0AD9toGBqXWlX-wCCIVobG9ACHrzWp5HpQmnxtN1tXEFIdfMv6xdFM5aKhcNLf8WEWRxueOdXvD697tB921bxWpxIFqy-pKiAsHK5hMc0JT9e1DLwqnsu6Qt2TzA69hI4mTyhfRZiemrRY82yUggCMyBjC1YjtZMjKem2g3CHmnFR-P5bGQ',
  'username': 'front-end-test'
}

export default secrets
