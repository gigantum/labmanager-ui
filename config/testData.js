import developerConfig

export const testData = {
  createLabbookMutation:{
    labbookName: 'ui-test-automated',
    'owner': 'gigantum_test'
  }
}
