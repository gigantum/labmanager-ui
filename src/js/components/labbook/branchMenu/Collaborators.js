//vendor
import React, { Component } from 'react'
import classNames from 'classnames'
//Mutations
import AddCollaboratorMutation from 'Mutations/AddCollaboratorMutation'
import DeleteCollaboratorMutation from 'Mutations/DeleteCollaboratorMutation'
//components
import ButtonLoader from 'Components/shared/ButtonLoader'
//store
import store from 'JS/redux/store'

export default class DeleteLabbook extends Component {
  constructor(props){
  	super(props);
    const {
      labbookName,
      owner
    } = props

  	this.state = {
      labbookName,
      owner,
      'addCollaboratorButtonDisabled': false,
      'newCollaborator': '',
      'buttonLoaderRemoveCollaborator': {},
      'buttonLoaderAddCollaborator': ''
    };

    this._addCollaborator = this._addCollaborator.bind(this)
    this._removeCollaborator = this._removeCollaborator.bind(this)
  }

  componentDidMount() {
    let buttonLoaderRemoveCollaborator = {}

<<<<<<< HEAD
    this.props.collaborators.forEach((collaborator) => {
=======
    this.props.collaborators.map((collaborator) => {
>>>>>>> integration
      buttonLoaderRemoveCollaborator[collaborator] = ''
    })

    this.setState({buttonLoaderRemoveCollaborator})
  }
  /**
  *  @param {event} evt
  *  sets state of Collaborators
  *  @return {}
  */
  _addCollaborator(evt) {
    const {
      labbookName,
      owner,
      newCollaborator } = this.state
    if ((evt.type === 'click') || (evt.key === "Enter")) {
      //waiting for backend updates
      this.setState({addCollaboratorButtonDisabled: true, buttonLoaderAddCollaborator: 'loading'})

      AddCollaboratorMutation(
        labbookName,
        owner,
        newCollaborator,
        (response, error) => {

          let {buttonLoaderRemoveCollaborator} = this.state

          buttonLoaderRemoveCollaborator[newCollaborator] = ''

          this.setState({ newCollaborator: '', addCollaboratorButtonDisabled: false, buttonLoaderRemoveCollaborator})

          this.inputTitle.value = ''

          if (error) {
            console.log(error)
            store.dispatch({
              type: 'ERROR_MESSAGE',
              payload: {
                message: `Could not add collaborator`,
                messageBody: error
              }
            })

            this.setState({buttonLoaderAddCollaborator: 'error'})
          } else {
            this.setState({buttonLoaderAddCollaborator: 'finished'})
            setTimeout(()=>{
              this.setState({buttonLoaderAddCollaborator: ''})
            }, 2000)
          }
        }

      )
    } else {
      this.setState({ newCollaborator: evt.target.value })
    }
  }
  /**
  *  @param {object} params
  *  deletes collaborators using mutation
  *  @return {}
  */
  _removeCollaborator(evt, params) {

    const {collaborator, button} = params

    button.disabled = true;

    let {buttonLoaderRemoveCollaborator} = this.state

    buttonLoaderRemoveCollaborator[collaborator] = 'loading'

    this.setState({buttonLoaderRemoveCollaborator})

    DeleteCollaboratorMutation(
      this.state.labbookName,
      this.state.owner,
      collaborator,
      (response, error) => {
        this.refs[collaborator] && this.refs[collaborator].classList.remove('loading');
        if(button){
          button.disabled = false;
        }
        if (error) {
          store.dispatch({
            type: 'ERROR_MESSAGE',
            payload: {
              message: `Could not remove collaborator`,
              messageBody: error
            }
          })

          buttonLoaderRemoveCollaborator[collaborator] = 'error'

          this.setState({buttonLoaderRemoveCollaborator})
        }else{
          buttonLoaderRemoveCollaborator[collaborator] = 'finished'

          this.setState({buttonLoaderRemoveCollaborator})
        }

        setTimeout(() => {
          buttonLoaderRemoveCollaborator[collaborator] = ''

          this.setState({buttonLoaderRemoveCollaborator})
        }, 2000)
      }
    )
  }

  render(){

    return(
      <div className="">

        <h4
          className="BranchModal__header">Manage Collaborators</h4>
        <hr />

        <div className="BranchMenu__collaborator-container">

          <div className="BranchMenu__add">

            <input
              ref={el => this.inputTitle = el}
              onChange={(evt) => this._addCollaborator(evt)}
              onKeyUp={(evt) => this._addCollaborator(evt)}
              className="BranchMenu__add-collaborators"
              type="text"
              placeholder="Add Collaborator" />


            <ButtonLoader
              ref="addCollaborator"
              buttonState={this.state.buttonLoaderAddCollaborator}
              buttonText={""}
              className="BranchMenu__add-button"
              params={{}}
              buttonDisabled={this.state.addCollaboratorButtonDisabled || !this.state.newCollaborator.length}
              clicked={this._addCollaborator}
            />

          </div>

          <div className="BranchMenu__collaborators">

          <h5>Collaborators</h5>

            <div className="BranchMenu__collaborators-list-container">

              {this.props.collaborators &&

                <ul className="BranchMenu__collaborators-list">
                  {
                    this.props.collaborators.map((collaborator) => {

                      const collaboratorItemCSS = classNames({
                        "BranchMenu__collaborator--item-me": collaborator === localStorage.getItem('username'),
                        "BranchMenu__collaborator--item": !(collaborator === localStorage.getItem('username'))
                      })

                      return (

                        <li
                          key={collaborator}
                          ref={collaborator}
                          className={collaboratorItemCSS}>

                            <div>{collaborator}</div>

                            <ButtonLoader
                              ref={collaborator}
                              buttonState={this.state.buttonLoaderRemoveCollaborator[collaborator]}
                              buttonText={""}
                              className="BranchMenu__collaborator-button"
                              params={{collaborator, button: this}}
                              buttonDisabled={collaborator === localStorage.getItem('username')}
                              clicked={this._removeCollaborator}
                            />

                        </li>)

                    })

                  }

                </ul>

              }

            </div>
          </div>
        </div>

      </div>
    )
  }
}
