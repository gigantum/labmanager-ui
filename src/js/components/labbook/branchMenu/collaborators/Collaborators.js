
import {QueryRenderer, graphql} from 'react-relay'
import React, { Component } from 'react'
import classNames from 'classnames'
//environment
import environment from 'JS/createRelayEnvironment'
//components
import CollaboratorsModal from './CollaboratorsModal'
//store
import store from 'JS/redux/store'

export const CollaboratorsQuery =  graphql`
  query CollaboratorsQuery($name: String!, $owner: String!){
    labbook(name: $name, owner: $owner){
      collaborators
      canManageCollaborators
    }
  }`


  export default class CollaboratorButton extends Component {
    constructor(props){
    	super(props);
    	this.state = {
        collaboratorModalVisible: false,
        canClickCollaborators: false,
        sessionValid: true,
      };

      this._toggleCollaborators = this._toggleCollaborators.bind(this)
    }

    _toggleCollaborators(){
      if(navigator.onLine){
        if( this.state.sessionValid ){
          this.setState({collaboratorModalVisible: !this.state.collaboratorModalVisible})
        } else{
          this.props.auth.renewToken(true, ()=>{
            this.props.showLoginPrompt();
          }, ()=>{
            this.setState({collaboratorModalVisible: !this.state.collaboratorModalVisible})
          });
        }
      } else {
        this.props.showLoginPrompt();
      }
    }

    _getCollaboratorList(collaborators, collaboratorFilteredArr){
      let lastParsedIndex
      let collaboratorSubText = collaboratorFilteredArr ? collaboratorFilteredArr.join(', ') : ''

      if(collaboratorSubText.length > 18 && collaboratorSubText.length){

        collaboratorSubText = collaboratorSubText.slice(0,18)

        lastParsedIndex = collaboratorSubText.split(', ').length -1;

        let lastParsed = collaboratorSubText.split(', ')[lastParsedIndex]

        if(collaborators[lastParsedIndex] !== lastParsed){

          lastParsedIndex--;

          collaboratorSubText = collaboratorSubText.split(', ')
          collaboratorSubText.pop();
          collaboratorSubText = collaboratorSubText.join(', ')

        }

        collaboratorSubText += `...+${collaboratorFilteredArr.length - lastParsedIndex - 1}`

      }
      return collaboratorSubText
    }

    render(){
      let self = this
      const {labbookName, owner} = this.props
      let {collaborators, canManageCollaborators} = store.getState().collaborators;
      collaborators = collaborators && collaborators[labbookName];
      canManageCollaborators = canManageCollaborators && canManageCollaborators[labbookName];
      this.props.checkSessionIsValid().then((res) => {
        if(res.data && res.data.userIdentity && res.data.userIdentity.isSessionValid) {
          if(this.state.sessionValid === false){
            this.setState({sessionValid: true})
          }
        } else {
          if(this.state.sessionValid === true){
            this.setState({sessionValid: false})
          }
        }
      })
      return(
        <QueryRenderer
          query={CollaboratorsQuery}
          environment={environment}
          variables={{
            name: labbookName,
            owner: owner
          }}
          render={({props, error})=> {
              if(props){
                const {labbook} = props
                store.dispatch({
                  type: 'SET_COLLABORATORS',
                  payload: {
                    collaborators: {
                      [labbookName]: labbook.collaborators
                    }
                  }
                })

                store.dispatch({
                  type: 'SET_CANMANAGECOLLABORATORS',
                  payload: {
                    canManageCollaborators: {
                      [labbookName]: labbook.canManageCollaborators
                    }
                  }
                })
                const collaboratorButtonCSS = classNames({
                  'disabled': !labbook.canManageCollaborators && this.state.sessionValid,
                  'BranchMenu__item--flat-button':  true
                })

                const collaboratorCSS = classNames({
                  'BranchMenu__item--collaborators': true,
                  'disabled': !labbook.canManageCollaborators && this.state.sessionValid,
                })

                const collaboratorFilteredArr = labbook.collaborators && labbook.collaborators.filter((name)=>{
                  return name !== owner
                })

                const collaboratorNames = self._getCollaboratorList(labbook.collaborators, collaboratorFilteredArr)

                return(

                  <li className={collaboratorCSS}
                    >
                    <button
                      onClick={() => this._toggleCollaborators()}
                      className={collaboratorButtonCSS}>
                      Collaborators

                      <p className="BranchMenu__collaborator-names">{collaboratorNames}</p>

                    </button>

                    {
                      this.state.collaboratorModalVisible &&
                        <CollaboratorsModal
                          key="CollaboratorsModal"
                          ref="collaborators"
                          collaborators={labbook.collaborators}
                          owner={owner}
                          labbookName={labbookName}
                          toggleCollaborators={this._toggleCollaborators}
                        />

                    }
                  </li>
                )
              }else if(collaborators !== null){
                const collaboratorButtonCSS = classNames({
                  'disabled': !canManageCollaborators && this.state.sessionValid,
                  'BranchMenu__item--flat-button':  true
                })

                const collaboratorCSS = classNames({
                  'BranchMenu__item--collaborators': true,
                  'disabled': !canManageCollaborators  && this.state.sessionValid
                })

                const collaboratorFilteredArr = collaborators && collaborators.filter((name)=>{
                  return name !== owner
                })

                const collaboratorNames = self._getCollaboratorList(collaborators, collaboratorFilteredArr)

                return(

                  <li className={collaboratorCSS}
                    >
                    <button
                      onClick={() => this._toggleCollaborators()}
                      className={collaboratorButtonCSS}>
                      Collaborators

                      <p className="BranchMenu__collaborator-names">{collaboratorNames}</p>

                    </button>

                    {
                      this.state.collaboratorModalVisible &&

                        [<CollaboratorsModal
                          key="CollaboratorsModal"
                          ref="collaborators"
                          collaborators={collaborators}
                          owner={owner}
                          labbookName={labbookName}
                          toggleCollaborators={this._toggleCollaborators}
                        />,
                        <div
                          key="CollaboratorsModal__cover"
                          className="modal__cover--nested"></div>
                      ]

                    }
                  </li>
                )
              } else{
                return(
                  <li className="BranchMenu__item--collaborators disabled"
                    >
                    <button
                      onClick={() => this.props.showLoginPrompt()}
                      className="BranchMenu__item--flat-button disabled">Collaborators</button>
                  </li>
                )
              }
          }}

        />

      )

    }
  }
