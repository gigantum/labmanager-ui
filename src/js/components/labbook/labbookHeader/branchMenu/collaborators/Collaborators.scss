@import '~Styles/_imports.scss';




.BranchMenu__collaborators-list-container {
  height: 200px;
  border: 1px solid $background-color;
  ul {
    @include flex(flex-start, column);
    flex-wrap: wrap;
    padding: 0 13px;
    overflow: auto;
  }
}
