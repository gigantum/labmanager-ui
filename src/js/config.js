import tips from './tips'

const CONFIG = {
  'api': process.env.NODE_ENV,
  'navigation_items': [
    {id:'overview', name: 'Overview'},
    {id:'activity', name: 'Activity', 'fragment': '...Activity_labbook'},
    {id:'environment', name: 'Environment', 'fragment': '...Environment_labbook'},
    {id:'code', name: 'Code'},
    {id:'inputData', name: 'Input Data'},
    {id:'outputData', name: 'Output Data'}
  ],
  'modalNav': [
    {'id': 'createLabook', 'description': 'Title & Description'},
    {'id': 'selectBaseImage', 'description': 'Base Image'},
    {'id': 'selectDevelopmentEnvironment', 'description': 'Dev Environment'},
    {'id': 'addEnvironmentPackage', 'description': 'Add Dependencies'},
    {'id': 'successMessage', 'description': 'Success'}
  ],
  'months': [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec'
  ],
  'fileBrowser':{
    'excludedFiles': [
      'DS_Store',
      'zip',
      'lbk',
      'pyc',
      'gitkeep'
    ]
  },

  containerStatus:{
    canEditEnvironment: (status) => {
      return (status === 'Stopped') || (status === 'Rebuild')
    }
  },
  userAPI: {
    getUsersQueryString: (userInput) => {
      const sanitizedUserInput = userInput.replace(/-/g, ' ')
      const apiURL = `https://m9eq4m3z0f.execute-api.us-east-1.amazonaws.com/prod?q=${sanitizedUserInput}*&q.options={fields: ['username^5','name']}&size=10`

      return encodeURI(apiURL)
    },
    getUserEmailQueryString: (email) => {
      const apiURL = `https://m9eq4m3z0f.execute-api.us-east-1.amazonaws.com/prod?q=${email}&q.options={fields: ['email']}&size=10`

      return encodeURI(apiURL)
    }
  },
  getToolTipText: (section) =>{
    return tips[section]
  },
  demoHostName: 'try.gigantum.com',
}

export default CONFIG
