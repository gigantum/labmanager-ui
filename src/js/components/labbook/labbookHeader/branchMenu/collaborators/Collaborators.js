import { QueryRenderer, graphql } from 'react-relay';
import React, { Component } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';

// environment
import environment from 'JS/createRelayEnvironment';
// components
import CollaboratorsModal from './CollaboratorsModal';
// store
import store from 'JS/redux/store';
import { setCollaborators, setCanManageCollaborators } from 'JS/redux/reducers/labbook/branchMenu/collaborators/collaborators';
// assets
import './Collaborators.scss';

export const CollaboratorsQuery = graphql`
  query CollaboratorsQuery($name: String!, $owner: String!){
    labbook(name: $name, owner: $owner){
      collaborators
      canManageCollaborators
    }
  }`;


class CollaboratorButton extends Component {
  constructor(props) {
    	super(props);
    	this.state = {
      collaboratorModalVisible: false,
      canClickCollaborators: false,
      sessionValid: true,
    };

    this._toggleCollaborators = this._toggleCollaborators.bind(this);
  }

  static getDerivedStateFromProps(nextProps, nextState) {
    return nextProps.checkSessionIsValid().then((res) => {
      if (res.data && res.data.userIdentity && res.data.userIdentity.isSessionValid) {
        return {
          ...nextState,
          sessionValid: true,
        };
      }
      return {
        ...nextState,
        sessionValid: false,
      };
    });
  }

  componentDidMount() {
    this.props.setCollaborators({ [this.props.labbookName]: this.collaborators });
    this.props.setCanManageCollaborators({ [this.props.labbookName]: this.canManageCollaborators });
  }

  _toggleCollaborators() {
    if (navigator.onLine) {
      if (this.state.sessionValid) {
        this.setState({ collaboratorModalVisible: !this.state.collaboratorModalVisible });
      } else {
        this.props.auth.renewToken(true, () => {
          this.props.showLoginPrompt();
        }, () => {
          this.setState({ collaboratorModalVisible: !this.state.collaboratorModalVisible });
        });
      }
    } else {
      this.props.showLoginPrompt();
    }
  }

  _getCollaboratorList(collaborators, collaboratorFilteredArr) {
    let lastParsedIndex;
    let collaboratorSubText = collaboratorFilteredArr ? collaboratorFilteredArr.join(', ') : '';

    if (collaboratorSubText.length > 18 && collaboratorSubText.length) {
      collaboratorSubText = collaboratorSubText.slice(0, 18);

      lastParsedIndex = collaboratorSubText.split(', ').length - 1;

      const lastParsed = collaboratorSubText.split(', ')[lastParsedIndex];

      if (collaborators[lastParsedIndex] !== lastParsed) {
        lastParsedIndex--;

        collaboratorSubText = collaboratorSubText.split(', ');
        collaboratorSubText.pop();
        collaboratorSubText = collaboratorSubText.join(', ');
      }

      collaboratorSubText += `...+${collaboratorFilteredArr.length - lastParsedIndex - 1}`;
    }
    return collaboratorSubText;
  }

  render() {
    // console.log('here in render')
    const self = this;
    const { labbookName, owner } = this.props;
    let { collaborators, canManageCollaborators } = store.getState().collaborators;
    collaborators = collaborators && collaborators[labbookName];
    canManageCollaborators = canManageCollaborators && canManageCollaborators[labbookName];
    return (
      <QueryRenderer
        query={CollaboratorsQuery}
        environment={environment}
        variables={{
            name: labbookName,
            owner,
          }}
        render={({ props, error }) => {
              if (props) {
                const { labbook } = props;

                this.canManageCollaborators = labbook.canManageCollaborators;
                this.collaborators = labbook.collaborators;

                const collaboratorButtonCSS = classNames({
                    disabled: !labbook.canManageCollaborators && this.state.sessionValid,
                    'BranchMenu__btn--flat': true,
                  }),

                  collaboratorCSS = classNames({
                    'BranchMenu__item BranchMenu__item--collaborators': true,
                    disabled: !labbook.canManageCollaborators && this.state.sessionValid,
                  }),

                  collaboratorFilteredArr = labbook.collaborators && labbook.collaborators.filter(name => name !== owner);

                const collaboratorNames = self._getCollaboratorList(labbook.collaborators, collaboratorFilteredArr);

                return (

                  <li className={collaboratorCSS}>
                    <button
                      onClick={() => this._toggleCollaborators()}
                      className={collaboratorButtonCSS}
                    >
                      Collaborators

                      <p className="BranchMenu__collaborator-names">{collaboratorNames}</p>

                    </button>

                    {
                      this.state.collaboratorModalVisible &&
                        <CollaboratorsModal
                          key="CollaboratorsModal"
                          ref="collaborators"
                          collaborators={labbook.collaborators}
                          owner={owner}
                          labbookName={labbookName}
                          toggleCollaborators={this._toggleCollaborators}
                        />

                    }
                  </li>
                );
              } else if (collaborators !== null) {
                const collaboratorButtonCSS = classNames({
                  disabled: !canManageCollaborators && this.state.sessionValid,
                  'BranchMenu__item--flat-button': true,
                });

                const collaboratorCSS = classNames({
                  'BranchMenu__item--collaborators': true,
                  disabled: !canManageCollaborators && this.state.sessionValid,
                });

                const collaboratorFilteredArr = collaborators && collaborators.filter(name => name !== owner);

                const collaboratorNames = self._getCollaboratorList(collaborators, collaboratorFilteredArr);

                return (

                  <li className={collaboratorCSS}>
                    <button
                      onClick={() => this._toggleCollaborators()}
                      className={collaboratorButtonCSS}
                    >
                      Collaborators

                      <p className="BranchMenu__collaborator-names">{collaboratorNames}</p>

                    </button>

                    {
                      this.state.collaboratorModalVisible &&

                        [<CollaboratorsModal
                          key="CollaboratorsModal"
                          ref="collaborators"
                          collaborators={collaborators}
                          owner={owner}
                          labbookName={labbookName}
                          toggleCollaborators={this._toggleCollaborators}
                        />,
                          <div
                            key="CollaboratorsModal__cover"
                            className="modal__cover--nested"
                          />,
                      ]

                    }
                  </li>
                );
              }
                return (
                  <li className="BranchMenu__item--collaborators disabled">
                    <button
                      onClick={() => this.props.showLoginPrompt()}
                      className="BranchMenu__item--flat-button disabled"
                    >Collaborators
                    </button>
                  </li>
                );
          }}
      />

    );
  }
}

const mapStateToProps = (state, ownProps) => ({

});

const mapDispatchToProps = dispatch => ({
  setCollaborators,
  setCanManageCollaborators,
});

export default connect(mapStateToProps, mapDispatchToProps)(CollaboratorButton);
