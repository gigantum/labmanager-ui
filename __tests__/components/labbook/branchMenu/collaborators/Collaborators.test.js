
      import React from 'react'
      import renderer from 'react-test-renderer';
      import {mount} from 'enzyme'
      import Collaborators from 'Components/labbook/branchMenu/collaborators/Collaborators';

      import json from './__relaydata__/Collaborators.json'

      import relayTestingUtils from 'relay-testing-utils'

      import Promise from 'promise'

      import config from '../../../../config'


      let promise = new Promise(resolve => {resolve(true)})
      const {owner, labbookName} = config



      const showLoginPrompt = () =>{

      }

      describe('Test Collaborators', () => {
        it('renders valid', () => {

          let checkSessionIsValid = (isValid) => {
            return promise.then(isValid => true);
          }

          let fixtures = {
              key: "test_collaborators",
              owner,
              labbookName,
              checkSessionIsValid,
              showLoginPrompt
          }

          const wrapper = renderer.create(

             relayTestingUtils.relayWrap(
               <Collaborators  {...fixtures} />, {}, json.data.collaborators
             )

          );

          const tree = wrapper.toJSON()

          expect(tree).toMatchSnapshot()
       })


       it('renders invalid', () => {

         let checkSessionIsValid = (isValid) => {
           return promise.then(isValid => false);
         }

         let fixtures = {
             key: "test_collaborators",
             owner,
             labbookName,
             checkSessionIsValid,
             showLoginPrompt
         }

         const wrapper = renderer.create(

            relayTestingUtils.relayWrap(
              <Collaborators  {...fixtures} />, {}, json.data.collaborators
            )

         );

         const tree = wrapper.toJSON()

         expect(tree).toMatchSnapshot()
      })

      })
