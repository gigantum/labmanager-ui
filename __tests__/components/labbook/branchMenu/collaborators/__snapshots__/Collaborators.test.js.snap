// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Collaborators renders invalid 1`] = `
<li
  className="BranchMenu__item--collaborators disabled"
>
  <button
    className="BranchMenu__item--flat-button disabled"
    onClick={[Function]}
  >
    Collaborators
  </button>
</li>
`;

exports[`Test Collaborators renders valid 1`] = `
<li
  className="BranchMenu__item--collaborators disabled"
>
  <button
    className="BranchMenu__item--flat-button disabled"
    onClick={[Function]}
  >
    Collaborators
  </button>
</li>
`;
